### DISCUSSION TITLE
`vcpkg_from_git` strips away `.git` folder?

### DISCUSSION THREAD

**Discussion by thomthom** [microsoft/vcpkg#21464] [Q&A] [answered]
I'm making a vcpkg package of a CMake project. The CMake project picks up the current git commit hash and makes it available to the build in order to provide some context to when a binary was created.

When I'm working on making this into a vcpkg I'm finding that `vcpkg_from_git` fetches the content into `vcpkg\buildtrees` without keeping the `.git` directory, so that I'm not able to query for the Git hash.

Is there another way I get obtain the git hash? Can I pass it along from `portfile.cmake`?

**Comment by mathisloge** [comment]
Just to extend this discussion a bit:
The .git dir is also necessary for https://bitbucket.org/chromiumembedded/cef to be built and that IIRC mainly hinders its integration into vcpkg

**Comment by thomthom** [comment, marked as answer]
I'm curious to why `.git` folder is stripped. `vcpkg_from_git` is making a call to `git clone`, right? So is that done in a temp location and the content copied without `.git`?

Atm, since I only need the git ref, I found I was able to forward the ref from the `port.cmake` file itself:

```
vcpkg_configure_cmake(
  SOURCE_PATH "${SOURCE_PATH}"
  PREFER_NINJA
  OPTIONS
    -DMY_LIB_GIT_REF=${GIT_SHORT_REF}
)
```

In in my library I make a check for that before I attempt to use `git` itself to obtain the ref.

**Reply by thomthom** [reply]
hm... I got an unexpected ref on macOS than on Windows. I setup my lib to pick the ref from git if `MY_LIB_GIT_REF` isn't set. That worked as expected on Windows.

On macOS I got the ref for the vcpkg submodule. So it appear the ref wasn't passed on in my macOS environment.
